GitUI/BranchTreePanel/RepoObjectsTree.cs

```diff
@@ -529,6 +504,49 @@ List<TreeNode> SearchTree(string text, IEnumerable<TreeNode> nodes)
             }
         }
 
+        /// <summary>
+        /// A function to see if the node text or full path of a node (for branch node) is matching with search text
+        /// </summary>
+        /// <param name="nodeText">nodeText/full path</param>
+        /// <param name="searchText">text entered in search box</param>
+        /// <param name="isFullMatch">is full match found</param>
+        /// <param name="matchIndex">index of the closest match found</param>
+        /// <param name="ret">List of matching nodes</param>
+        /// <returns>true if match found</returns>
+        private bool searchForMatchingText(string nodeText, string searchText, ref bool isFullMatch, ref int matchIndex, List<TreeNode> ret)
```

Comment:
Use a local method, no need to expose to other methods in this module.
Then it would not be required to use ref  - but I suggest keeping that as it improves readability (that is the bad part of local functions, they have access to all method variables). Local methods should be static if reasonable.

Similar for rotate

Reply:
I haven't used local methods before. It's an interesting feature.
To improve readability, I will retain all the method parameters including ref as before. 
There is no reason for both of these local methods to be static. Changes made to variables inside the function are required in the containing function also. 
Is my understanding correct?

Reply:
If the local method is static, it means it cannot change the main method variables, so that is a good documentation.
Visual Studio will suggest this if the function has no side effects.

Local methods is something I have missed since working with Pascal a long time ago. Not much else I miss from Pascal though.

Reply:
Thank you for your inputs. I have moved the functions to local methods.